packages/grpc-native-core/src/server.js

```diff
@@ -598,8 +608,9 @@ function handleUnary(call, handler, metadata) {
       } else {
         sendUnaryResponse(call, value, handler.serialize, trailer, flags);
       }
+      asyncResource.destroy();
```

Comment:
Can you elaborate on why it is okay that the destroy doesn't get called on all paths?

Reply:
It's not okay 🙃 I've added additional calls to `destroy` if the user function is never called.